pkg/notifications/notifications_integration_test.go

```diff
@@ -93,6 +93,84 @@ func TestNotificationMutationDestination(t *testing.T) {
 	require.NotNil(t, deleteResult)
 }
 
+func TestNotificationMutationDestinationByName(t *testing.T) {
+	t.Parallel()
+
+	n := newIntegrationTestClient(t)
+
+	accountID, err := mock.GetTestAccountID()
+	if err != nil {
+		t.Skipf("%s", err)
+	}
+
+	// Create a destination to work with in this test
+	testIntegrationDestinationNameRandStr := mock.RandSeq(5)
+	destination := AiNotificationsDestinationInput{}
+	destination.Type = AiNotificationsDestinationTypeTypes.WEBHOOK
+	destination.Properties = []AiNotificationsPropertyInput{
+		{
+			Key:          "url",
+			Value:        "https://webhook.site/94193c01-4a81-4782-8f1b-554d5230395b",
+			Label:        "",
+			DisplayValue: "",
+		},
+	}
+	destination.Auth = &AiNotificationsCredentialsInput{
+		Type: AiNotificationsAuthTypeTypes.TOKEN,
+		Token: AiNotificationsTokenAuthInput{
+			Token:  "Token",
+			Prefix: "Bearer",
+		},
+	}
+	destination.Name = fmt.Sprintf("test-notifications-destination-%s", testIntegrationDestinationNameRandStr)
+
+	// Test: Create
+	createResult, err := n.AiNotificationsCreateDestination(accountID, destination)
+	require.NoError(t, err)
+	require.NotNil(t, createResult)
+	require.NotEmpty(t, createResult.Destination.Auth)
+	require.Equal(t, ai.AiNotificationsAuthType("TOKEN"), createResult.Destination.Auth.AuthType)
+
+	// Test: Get Destination by name
+	filtersByName := ai.AiNotificationsDestinationFilter{
+		Name: createResult.Destination.Name,
+	}
+	sorter := AiNotificationsDestinationSorter{}
+	getDestinationByNameResult, err := n.GetDestinations(accountID, "", filtersByName, sorter)
+	require.NoError(t, err)
+	require.NotNil(t, getDestinationByNameResult)
+	assert.Equal(t, 1, getDestinationByNameResult.TotalCount)
+
+	// Test: Update Destination
+	updateDestination := AiNotificationsDestinationUpdate{}
+	updateDestination.Active = false
+	updateDestination.Properties = []AiNotificationsPropertyInput{
+		{
+			Key:          "url",
+			Value:        "https://webhook.site/94193c01-4a81-4782-8f1b-554d5230395b",
+			Label:        "",
+			DisplayValue: "",
+		},
+	}
+	updateDestination.Auth = &AiNotificationsCredentialsInput{
+		Type: AiNotificationsAuthTypeTypes.TOKEN,
+		Token: AiNotificationsTokenAuthInput{
+			Token:  "TokenUpdate",
+			Prefix: "BearerUpdate",
+		},
+	}
+	updateDestination.Name = fmt.Sprintf("test-notifications-update-destination-%s", testIntegrationDestinationNameRandStr)
+
+	updateDestinationResult, err := n.AiNotificationsUpdateDestination(accountID, updateDestination, createResult.Destination.ID)
+	require.NoError(t, err)
+	require.NotNil(t, updateDestinationResult)
+
+	// Test: Delete
+	deleteResult, err := n.AiNotificationsDeleteDestination(accountID, createResult.Destination.ID)
+	require.NoError(t, err)
+	require.NotNil(t, deleteResult)
```

Comment:
What if the test fails? Do we not remove the destination?

Reply:
Not sure there is a better way of handling this - also I checked other places in this repo and it doesn't seems anyone else is cleaning this. @pranav-new-relic do you know if there is a cleaning after the integration tests for all the resources that weren't deleted?

Reply:
I don't think so, but I think maybe what @IliaShurygin means is the usage of cleanup mechanisms like in the Workflows Integration Test (I found the following as an example that you had both written)

https://github.com/newrelic/newrelic-client-go/blob/main/pkg/workflows/workflows_integration_test.go#L519

Would you like to take a look and see if something similar can be added in the current test when there is an error thrown (and is unexpected, which would eventually lead to a failure)

Reply:
Actually, the cleanup is another call to the API (which originally failed in the test), so I'm not sure how this should help other than it has one more attempt to delete it...